pkg/rpc/rpcstatus/status_grpc.go

```diff
@@ -6,45 +6,171 @@
 package rpcstatus
 
 import (
+	"context"
+	"errors"
+	"fmt"
+
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
+
+	"storj.io/drpc/drpcerr"
 )
 
-// StatusCode is the type of status codes for grpc.
-type StatusCode = codes.Code
+// StatusCode is the type of status codes for drpc.
+type StatusCode uint64
 
 // These constants are all the rpc error codes.
 const (
-	OK                 = codes.OK
-	Canceled           = codes.Canceled
-	Unknown            = codes.Unknown
-	InvalidArgument    = codes.InvalidArgument
-	DeadlineExceeded   = codes.DeadlineExceeded
-	NotFound           = codes.NotFound
-	AlreadyExists      = codes.AlreadyExists
-	PermissionDenied   = codes.PermissionDenied
-	ResourceExhausted  = codes.ResourceExhausted
-	FailedPrecondition = codes.FailedPrecondition
-	Aborted            = codes.Aborted
-	OutOfRange         = codes.OutOfRange
-	Unimplemented      = codes.Unimplemented
-	Internal           = codes.Internal
-	Unavailable        = codes.Unavailable
-	DataLoss           = codes.DataLoss
-	Unauthenticated    = codes.Unauthenticated
+	Unknown StatusCode = iota
+	OK
+	Canceled
```

Comment:
is the reason we can't reuse the values because the value and the type both matter in places?

Reply:
I'm not sure why they are different for grpc and drpc. I tried to avoid changing the behavior for that case.

Reply:
oh just cause it's already deployed, okay, rats

Reply:
the only difference is that `Unknown` is the zero value rather than `Ok`.

Additionally,I don’t understand why this block of constant definitions needs to change.

Reply:
`status_grpc.go` now by default uses the same constants as `drpc`, but has conversions to GRPC codes via `GRPCStatus`. I could have made the conversion the other way around, but I don't think it would have made much difference.

Reply:
Because these are the values for `drpc` not `grpc`. I could separate the `drpc` constants to a separate file and remove the duplication of `StatusCode` and constants.

Reply:
The DRPC constants are in the other file. I’m confused.

I’m arguing that they aren’t duplicates, and leaving the type alias and constants in this file unchanged, even if they differ from the drpc values, is fine.

Reply:
Then the constants wouldn't match with DRPC responses. But, sure I could use grpc constants then move drpc constants to a separate file and convert from grpc to drpc.

Reply:
The numeric value of the constants already don’t match, and the drpc constants are already in `status_drpc.go`. I’m just trying to understand the motivation for changing this block of constants away from being aliases.

Reply:
The short version:
1. we need something as a source of truth (either grpc or drpc) for the statuses
2. we need to convert from source of truth to the other version

This means we need access to both constants: drpc and grpc. Currently the files are conditionally compiled, which means that we need to expose `drpc` constants to `grpc` somewhere. Then we need a way to convert from one `code` to another, which means we need the conversions.

Which one we use a source of truth, doesn't matter that much. We could use aliases, but then we would need to still expose `drpc` constants and implement the custom error type that is compatible with both `drpc` and `grpc`.

Reply:
Understood. Thanks!